x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleType.java

```diff
@@ -86,12 +89,38 @@ public String getWriteableName() {
                     actionMap.put(UnfollowAction.NAME, new UnfollowAction());
                     phase = new Phase(phase.getName(), phase.getMinimumAge(), actionMap);
                 }
+
+                if (shouldMigrateDataToTiers(phase)) {
+                    Map<String, LifecycleAction> actionMap = new HashMap<>(phase.getActions());
+                    actionMap.put(MigrateAction.NAME, new MigrateAction(true));
+                    phase = new Phase(phase.getName(), phase.getMinimumAge(), actionMap);
+                }
+
                 orderedPhases.add(phase);
             }
         }
         return orderedPhases;
     }
 
+    boolean shouldMigrateDataToTiers(Phase phase) {
+        AllocateAction allocateAction = (AllocateAction) phase.getActions().get(AllocateAction.NAME);
+        if (allocateAction != null) {
+            if (allocateAction.getExclude().isEmpty() == false ||
+                allocateAction.getInclude().isEmpty() == false ||
+                allocateAction.getRequire().isEmpty() == false) {
```

Comment:
This could use your `definesAllocationRules(allocateAction)` helper here instead of manually checking, though I think maybe that helper needs to use `isEmpty` rather than checking for `null` so that an allocate action with `"include": {}` doesn't trip the assertion